warehouse/query-core/src/main/java/datawave/query/jexl/visitors/whindex/SplitGeoWaveFunctionVisitor.java

```diff
@@ -41,13 +45,22 @@ public static <T extends JexlNode> T apply(T script, MetadataHelper metadataHelp
     @Override
     public Object visit(ASTFunctionNode node, Object data) {
         JexlArgumentDescriptor descriptor = JexlFunctionArgumentDescriptorFactory.F.getArgumentDescriptor(node);
+        Set<String> fields = Sets.newHashSet();
         if (descriptor instanceof GeoWaveFunctionsDescriptor.GeoWaveJexlArgumentDescriptor) {
-            Set<String> fields = null;
             try {
                 fields = descriptor.fields(metadataHelper, null);
-            } catch (TableNotFoundException | InstantiationException | IllegalAccessException | ExecutionException |
-                     MarkingFunctions.Exception e) {
-                LOGGER.debug("Unable to load datatypes from metadata table");
+            } catch (TableNotFoundException e) {
+                QueryException qe = new QueryException(DatawaveErrorCode.METADATA_TABLE_FETCH_ERROR, e);
+                LOGGER.error(qe);
+                throw new DatawaveFatalQueryException(qe);
+            } catch (InstantiationException | IllegalAccessException e) {
+                QueryException qe = new QueryException(DatawaveErrorCode.METADATA_TABLE_RECORD_FETCH_ERROR, e);
+                LOGGER.error(qe);
+                throw new DatawaveFatalQueryException(qe);
+            } catch (ExecutionException | MarkingFunctions.Exception e) {
+                QueryException qe = new QueryException(DatawaveErrorCode.UNKNOWN_SERVER_ERROR, e);
+                LOGGER.error(qe);
+                throw new DatawaveFatalQueryException(qe);
```

Comment:
I think I made this comment before, but why not put this exception handling in the `fields` method?  It seems like we're copying the same exception handling all over the place when we could probably just consolidate it by having it taken care of within the method.